site-builder/server/server.ts

```diff
@@ -0,0 +1,586 @@
+				};
+			}
+
+			console.log("Build successful");
+
+			let uploadResult: string | undefined;
+			let s3FolderUrl: string | undefined;
+			let s3FolderPath: string | undefined;
+			let error: Error | undefined;
+
+			try {
+				// Stream zip directly to S3 without saving to disk first
+				const zipFileName = `site-${timestamp}.zip`;
+				const uploadId = "site-archives"; // Folder name in the bucket
+
+				console.log("Creating and streaming zip archive directly to S3");
+
+				// This creates the zip and streams it directly to S3
+				uploadResult = await createZipAndUploadToS3(distDir, uploadId, zipFileName);
+
+				console.log(`Zip archive uploaded to: ${uploadResult}`);
+
+				// If requested, also upload dist contents to S3 folder
+				if (uploadToS3Folder) {
+					console.log("Additionally uploading dist contents to S3 folder");
+
+					// Use community slug as part of the path for better organization
+					const folderPath = `sites/${communitySlug}`;
+					console.log({
+						msg: `Uploading dist contents to S3 folder`,
+						distDir,
+						folderPath,
+						timestamp,
+					});
+					s3FolderUrl = await uploadDirectoryToS3(distDir, folderPath, timestamp);
+					s3FolderPath = `${folderPath}/${timestamp}`;
+
+					console.log(`Dist contents uploaded to: ${s3FolderUrl}`);
+				}
+
+				console.log("Process completed successfully");
+
+				return {
+					status: 200,
+					body: {
+						success: true,
+						message: "Site built and uploaded successfully",
+						url: uploadResult,
+						timestamp: timestamp,
+						// We can't determine the exact file size without saving to disk
+						// or collecting that data during archiving/upload, so we provide estimated values
+						fileSize: 0, // Required by the API contract, but we don't know the exact size
+						fileSizeFormatted: "Unknown (streaming upload)",
+						...(uploadToS3Folder && {
+							s3FolderUrl,
+							s3FolderPath,
+						}),
+					},
+				};
+			} catch (err) {
+				console.error("Error during build and upload process:", err);
+				error = err as Error;
+
+				return {
+					status: 500,
+					body: {
+						success: false,
+						message: error.message || "An unknown error occurred",
+						...(uploadResult && { url: uploadResult }),
+						...(s3FolderUrl && { s3FolderUrl }),
+					},
+				};
+			}
+		} catch (err) {
+			console.error("Error during build and upload process:", err);
+			const error = err as Error;
+
+			return {
+				status: 500,
+				body: {
+					success: false,
+					message: error.message || "An unknown error occurred",
+				},
+			};
+		}
+	},
+	health: async () => {
+		return {
+			status: 200,
+			body: {
+				status: "ok",
+			},
+		};
+	},
+});
+
+// // Health check endpoint
+// app.get("/health", (c) => {
+// 	return c.json({ status: "ok" });
+// });
+
+app.all("*", async (c) => {
+	return fetchRequestHandler({
+		request: new Request(c.req.url, c.req.raw),
+		contract: siteBuilderApi,
+		router,
+		options: {
+			//
+		},
+	});
+});
```

Comment:
this is how you use ts-rest with `hono`, although i think this instantiates the entire route structure on every request, which is obviously a bit slow.